### PR TITLE
chore: Add support for PHP 8.3

### PR DESCRIPTION
**Pull request by TimWolla**
No changes to the code required.

----------------

Feature freeze was some time ago and the first RC will be released next week (https://wiki.php.net/todo/php83). It appears appropriate to test it in CI, which also requires the update to composer.json (unless you prefer `--ignore-platform-req=php+`?).

### CODE REVIEW

**Comment by TimWolla** [comment]
If this PR is merged, the PHP 8.3 status check should be set to “Required” in the repository config.

File: .github/workflows/quality-assurance.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     name: Quality Assurance
     runs-on: ubuntu-latest
     env:
-      php-version: '8.2'
+      php-version: '8.3'
```
Comment (TimWolla) [inline .github/workflows/quality-assurance.yml:8]: Do you prefer:

1. `PHP_CS_FIXER_IGNORE_ENV`
2. A revert of this change.
3. To wait until a compatible PHP CS Fixer version is available?

Comment (romm) [reply .github/workflows/quality-assurance.yml:8]: Already had this PHPCSFixer issue, you can add  `PHP_CS_FIXER_IGNORE_ENV: 1 # PHP8.3 remove when PHP-CS-Fixer fully supports PHP 8.3` in https://github.com/CuyZ/Valinor/blob/d608ee9bcf06b1d23925a1f70e12312573780095/.github/workflows/quality-assurance.yml#L9

**Comment by romm** [comment]
Perfect, thank you @TimWolla!

**Pull request merged**